### ISSUE TITLE
Add a value for the `autodoc_class_signature` variable to suppress the constructor signature entirely

### ISSUE THREAD

**Issue by stefanseefeld** [sphinx-doc/sphinx#9859]
The `autodoc_class_signature` variable is useful to indicate whether or not the constructor signature should be part of the class in documentation.
At present it supports two values: "mixed", to include the signature with the class name, and "separated", to print it separately. I would like to have the ability to suppress the signature entirely, given that the constructor (`__init__`) can be documented separately, in which case the information is redundant.
(In fact, I never understood why the constructor signature was bundled with the class name. Class and constructor are two distinct entities anyway...)

**Comment by goodmami** [comment]
I agree, and this can be accomplished in combination with `:exclude-members: __init__`, but since `autodoc_class_signature` is global it splits out documentation of `__init__` for all classes. I'm looking for this feature at the class-level rather than a global option for times when a class is not meant to be instantiated directly but I still want to describe the class attributes and methods (similar to the stdlib's [logging.Logger](https://docs.python.org/3/library/logging.html#logger-objects) class). E.g., if my Python source has this:

```python
class Example:
    """A class for examples."""

    def __init__(self, text: str, _metadata: dict[str, str]) -> None:
        self.text = text
        self._metadata = _metadata

    def get_meta(self, key: str) -> str:
        """Return the metadata value for *key*."""
        return self._metadata[key]
```

I want to document the class with its docstring but without the signature or init parameters. I've tried a few things:

Includes signature, docstring, and parameters:
```restructuredtext
.. autoclass:: Example
   :members:
```

Includes docstring and parameters, excludes signature:
```restructuredtext
.. autoclass:: Example()
   :members:
```

Excludes signature, docstring, and parameters; docstring manually given, attributes and methods must be manually added:
```restructuredtext
.. class:: Example

   A class for examples.

   .. autoattribute:: text
   .. automethod:: get_meta
```

I'd like something like this:

```restructuredtext
.. autoclass:: Example
   :class-signature: suppress
   :members:
```

(valid values of `:class-signature:` being `mixed`, `separated`, and `suppress`)
